ballerina/tests/22_variables.bal

```diff
@@ -0,0 +1,220 @@
+// Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/test;
+import ballerina/lang.value;
+@test:Config {
+    groups: ["variables", "input"]
+}
+isolated function testDuplicateInputVariables() returns error? {
+    string document = string`($userName:string, $userName:int){ greet (name: $userName) }`;
```

Comment:
Shouldn't the type names start with a capital letter? Check the other places as well. If this test is passing, there might be an error in the parsing logic.